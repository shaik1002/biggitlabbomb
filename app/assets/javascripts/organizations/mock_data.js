/* eslint-disable @gitlab/require-i18n-strings */

// This is temporary mock data that will be removed when completing https://gitlab.com/gitlab-org/gitlab/-/issues/454935

export const defaultOrganization = {
  id: 1,
  name: 'Default',
  web_url: '/-/organizations/default',
  avatar_url: null,
};
