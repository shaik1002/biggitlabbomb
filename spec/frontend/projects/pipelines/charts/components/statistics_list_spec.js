import { GlLink } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import Component from '~/projects/pipelines/charts/components/statistics_list.vue';
import { counts } from '../mock_data';

describe('StatisticsList', () => {
  let wrapper;

  const failedPipelinesLink = '/flightjs/Flight/-/pipelines?page=1&scope=all&status=failed';

  const findFailedPipelinesLink = () => wrapper.findComponent(GlLink);

  beforeEach(() => {
    wrapper = shallowMount(Component, {
      provide: {
        failedPipelinesLink,
      },
      propsData: {
        counts,
      },
    });
  });

  it('displays the counts data with labels', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays failed pipelines link', () => {
    expect(findFailedPipelinesLink().attributes('href')).toBe(failedPipelinesLink);
  });

  describe('with no failed piplines link', () => {
    beforeEach(() => {
      wrapper = shallowMount(Component, {
        propsData: {
          counts,
        },
      });
    });

    it('hides the failed pipelines link', () => {
      expect(findFailedPipelinesLink().exists()).toBe(false);
    });
  });
});
