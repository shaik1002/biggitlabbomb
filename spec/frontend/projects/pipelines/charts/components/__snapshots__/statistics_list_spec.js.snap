// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsList displays the counts data with labels 1`] = `
<div
  class="gl-display-flex gl-flex-wrap gl-gap-6 gl-mb-6"
>
  <div>
    <gl-single-stat-stub
      animationdecimalplaces="0"
      id="reference-0"
      shouldanimate="true"
      title="Total pipelines"
      titleiconclass=""
      unit=""
      usedelimiters="true"
      value="40,000"
      variant="muted"
    />
  </div>
  <div>
    <gl-single-stat-stub
      animationdecimalplaces="0"
      id="reference-1"
      shouldanimate="true"
      title="Success ratio"
      titleiconclass=""
      unit=""
      usedelimiters="true"
      value="50.00%"
      variant="muted"
    />
  </div>
  <div>
    <gl-single-stat-stub
      animationdecimalplaces="0"
      id="reference-2"
      shouldanimate="true"
      title="Successful pipelines"
      titleiconclass=""
      unit=""
      usedelimiters="true"
      value="20,000"
      variant="muted"
    />
  </div>
  <div>
    <gl-single-stat-stub
      animationdecimalplaces="0"
      id="reference-3"
      shouldanimate="true"
      title="Failed pipelines"
      titleiconclass=""
      unit=""
      usedelimiters="true"
      value="20,000"
      variant="muted"
    />
    <gl-link-stub
      class="gl-p-2"
      href="/flightjs/Flight/-/pipelines?page=1&scope=all&status=failed"
    >
      See details
    </gl-link-stub>
  </div>
</div>
`;
